### PR TITLE
Fix db name suffix parsing for replica

### PR DESCRIPTION
**Pull request by kschusternetformic**

### CODE REVIEW

**Comment by ostrolucky** [comment]
test required

File: ConnectionFactory.php [hunk 1]
```diff
@@ -75,6 +75,14 @@ public function createConnection(array $params, ?Configuration $config = null, ?
                 $params['dbname'] .= $params['dbname_suffix'];
             }
 
+            if (isset($params['replica']) && isset($params['replica']['dbname']) && isset($params['replica']['dbname_suffix'])) {
```
Comment (ostrolucky) [inline ConnectionFactory.php]:
```suggestion
            if (isset($params['replica']['dbname'], $params['replica']['dbname_suffix'])) {
```

File: ConnectionFactory.php [hunk 2]
```diff
@@ -75,6 +75,14 @@ public function createConnection(array $params, ?Configuration $config = null, ?
...
+                $params['replica']['dbname'] .= $params['replica']['dbname_suffix'];
+            }
+
+            if (isset($params['primary']) && isset($params['primary']['dbname']) && isset($params['primary']['dbname_suffix'])) {
```
Comment (ostrolucky) [inline ConnectionFactory.php]:
```suggestion
            if (isset($params['primary']['dbname'], $params['primary']['dbname_suffix'])) {
```

**Review by ostrolucky — CHANGES_REQUESTED** [review]

**Comment by kschusternetformic** [comment]
@ostrolucky Tests were added

**Pull request merged**